theseus_gui/src/pages/Settings.vue

```diff
@@ -1,7 +1,319 @@
+  saveButton.value.$el.style.opacity = 0
+}
+</script>
 
 <template>
   <div>
-    <p>Settings page</p>
+    <Button ref="saveButton" color="primary" class="save-btn" @click="saveSettings">Save</Button>
+    <Card class="theming">
+      <h2>Themes</h2>
+      <div class="setting-row">
+        <div class="description">
+          <h3>Color theme</h3>
+          <p>Change the global launcher color theme.</p>
+        </div>
+        <DropdownSelect
+          name="Theme dropdown"
+          :options="themeStore.themeOptions"
+          :default-value="themeStore.selectedTheme"
+          :model-value="themeStore.selectedTheme"
+          class="theme-dropdown"
+          @change="handleTheme"
+        />
+      </div>
+    </Card>
+    <Card class="settings-card">
+      <h2 class="settings-title">Java</h2>
+      <div class="settings-group">
+        <h3>Java 17 Location</h3>
+        <div class="toggle-setting">
+          <input
+            v-model="settings.java_globals.JAVA_17.path"
+            type="text"
+            class="input installation-input"
+            placeholder="/path/to/java17"
+          />
+          <span class="installation-buttons">
+            <Button @click="saveJavaPath">
+              <SearchIcon />
+              Auto Detect
+            </Button>
+            <Button @click="saveJavaPath">
+              <BrowseIcon />
+              Browse
+            </Button>
+            <Button @click="saveJavaPath">
+              <PlayIcon />
+              Test
+            </Button>
+          </span>
+        </div>
+      </div>
+      <div class="settings-group">
+        <h3>Java 8 Location</h3>
+        <div class="toggle-setting">
+          <input
+            v-model="settings.java_globals.JAVA_8.path"
+            type="text"
+            class="input installation-input"
+            placeholder="/path/to/java8"
+          />
+          <span class="installation-buttons">
+            <Button @click="saveJavaPath">
+              <SearchIcon />
+              Auto Detect
+            </Button>
+            <Button @click="saveJavaPath">
+              <BrowseIcon />
+              Browse
+            </Button>
+            <Button @click="saveJavaPath">
+              <PlayIcon />
+              Test
+            </Button>
+          </span>
+        </div>
+      </div>
+      <hr class="card-divider" />
+      <div class="settings-group">
+        <h3>Java Arguments</h3>
+        <input
+          v-model="settings.custom_java_args"
+          type="text"
+          class="input installation-input"
+          placeholder="-Xms10G -Xmx10G -XX:+UseG1GC -XX:+ParallelRefProcEnabled"
+        />
+      </div>
+      <div class="settings-group">
+        <h3>Environment Arguments</h3>
+        <input
+          v-model="settings.custom_env_args"
+          type="text"
+          class="input installation-input"
+          placeholder="-Xms10G -Xmx10G -XX:+UseG1GC -XX:+ParallelRefProcEnabled"
```

Comment:
the placeholder is wrong here, this isn't what the env args would look like. Instead, it should be something like `ENV1=TEST ENV2=TEST`